<a href="https://colab.research.google.com/github/badkarma-tech/Capestone-project/blob/main/Gender_Development_Index.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np

In [ ]:
import pandas as pd

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving Gender Development Index.csv to Gender Development Index.csv


In [ ]:
print("First 10 rows of the data")
df.head(10)

First 10 rows of the data


ISO3               Country Continent           Hemisphere  \
0  AFG           Afghanistan      Asia  Northern Hemisphere   
1  AGO                Angola    Africa  Southern Hemisphere   
2  ALB               Albania    Europe  Northern Hemisphere   
3  AND               Andorra    Europe  Northern Hemisphere   
4  ARE  United Arab Emirates      Asia  Northern Hemisphere   
5  ARG             Argentina   America  Southern Hemisphere   
6  ARM               Armenia      Asia  Northern Hemisphere   
7  ATG   Antigua and Barbuda   America  Northern Hemisphere   
8  AUS             Australia   Oceania  Southern Hemisphere   
9  AUT               Austria    Europe  Northern Hemisphere   

  Human Development Groups UNDP Developing Regions  HDI Rank (2021)  \
0                      Low                      SA            180.0   
1                   Medium                     SSA            148.0   
2                     High                     ECA             67.0   
3                Very High                     NaN             40.0   
4                Very High                      AS             26.0   
5                Very High                     LAC             47.0   
6                     High                     ECA             85.0   
7                     High                     LAC             71.0   
8                Very High                     NaN              5.0   
9                Very High                     NaN             25.0   

   Gender Development Index (1990)  Gender Development Index (1991)  \
0                            0.595                            0.588   
1                              NaN                              NaN   
2                            0.922                            0.921   
3                              NaN                              NaN   
4                            0.961                            0.958   
5                            0.982                            0.980   
6                              NaN                              NaN   
7                              NaN                              NaN   
8                            0.961                            0.961   
9                            0.940                            0.942   

   Gender Development Index (1992)  ...  Gender Development Index (2012)  \
0                            0.586  ...                            0.649   
1                              NaN  ...                            0.848   
2                            0.932  ...                            0.974   
3                              NaN  ...                              NaN   
4                            0.957  ...                            0.919   
5                            0.975  ...                            0.998   
6                              NaN  ...                            0.977   
7                              NaN  ...                              NaN   
8                            0.970  ...                            0.960   
9                            0.944  ...                            0.955   

   Gender Development Index (2013)  Gender Development Index (2014)  \
0                            0.659                            0.630   
1                            0.866                            0.884   
2                            0.968                            0.968   
3                              NaN                              NaN   
4                            0.920                            0.921   
5                            0.997                            0.995   
6                            0.982                            0.985   
7                              NaN                              NaN   
8                            0.961                            0.972   
9                            0.963                            0.972   

   Gender Development Index (2015)  Gender Development Index (2016)  \
0                            0.673                    

In [ ]:
print("Information")
df.info()


Information
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 39 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   ISO3                             195 non-null    object 
 1   Country                          195 non-null    object 
 2   Continent                        195 non-null    object 
 3   Hemisphere                       195 non-null    object 
 4   Human Development Groups         191 non-null    object 
 5   UNDP Developing Regions          151 non-null    object 
 6   HDI Rank (2021)                  191 non-null    float64
 7   Gender Development Index (1990)  111 non-null    float64
 8   Gender Development Index (1991)  111 non-null    float64
 9   Gender Development Index (1992)  111 non-null    float64
 10  Gender Development Index (1993)  112 non-null    float64
 11  Gender Development Index (1994)  112 non-null    float64
 12  Gender Dev

In [ ]:
print(df.describe())


       HDI Rank (2021)  Gender Development Index (1990)  \
count       191.000000                       111.000000   
mean         95.811518                         0.880387   
std          55.307333                         0.108414   
min           1.000000                         0.534000   
25%          48.500000                         0.820500   
50%          96.000000                         0.919000   
75%         143.500000                         0.959000   
max         191.000000                         1.072000   

       Gender Development Index (1991)  Gender Development Index (1992)  \
count                       111.000000                       111.000000   
mean                          0.882550                         0.885811   
std                           0.105711                         0.104148   
min                           0.540000                         0.545000   
25%                           0.823500                         0.826500   
50%               

In [ ]:
print(df.isnull().sum())



ISO3                                0
Country                             0
Continent                           0
Hemisphere                          0
Human Development Groups            4
UNDP Developing Regions            44
HDI Rank (2021)                     4
Gender Development Index (1990)    84
Gender Development Index (1991)    84
Gender Development Index (1992)    84
Gender Development Index (1993)    83
Gender Development Index (1994)    83
Gender Development Index (1995)    65
Gender Development Index (1996)    65
Gender Development Index (1997)    65
Gender Development Index (1998)    64
Gender Development Index (1999)    58
Gender Development Index (2000)    41
Gender Development Index (2001)    40
Gender Development Index (2002)    38
Gender Development Index (2003)    35
Gender Development Index (2004)    33
Gender Development Index (2005)    29
Gender Development Index (2006)    28
Gender Development Index (2007)    28
Gender Development Index (2008)    28
Gender Devel

In [ ]:
## Get the shape
shape = df.shape
num_rows, num_columns = shape

print(f"Number of Rows: {num_rows}")
print(f"Number of Columns: {num_columns}")

Number of Rows: 195
Number of Columns: 39


In [ ]:
# Calculate the correlation matrix for selected columns
correlation_matrix = df[['HDI Rank (2021)', 'Gender Development Index (2021)']].corr()
print(correlation_matrix)

                                 HDI Rank (2021)  \
HDI Rank (2021)                          1.00000   
Gender Development Index (2021)         -0.59408   

                                 Gender Development Index (2021)  
HDI Rank (2021)                                         -0.59408  
Gender Development Index (2021)                          1.00000  


In [ ]:
# Calculate the mean, median, and standard deviation for a specific column
mean_gdi_2021 = df['Gender Development Index (2021)'].mean()
median_gdi_2021 = df['Gender Development Index (2021)'].median()
std_gdi_2021 = df['Gender Development Index (2021)'].std()
print("\nMean GDI (2021):", mean_gdi_2021)
print("Median GDI (2021):", median_gdi_2021)
print("Standard Deviation GDI (2021):", std_gdi_2021)


Mean GDI (2021): 0.9472848837209302
Median GDI (2021): 0.967
Standard Deviation GDI (2021): 0.06590579137991819


In [ ]:
# Count the number of unique values in a column
unique_continents = df['Continent'].nunique()
print("\nNumber of Unique Continents:", unique_continents)


Number of Unique Continents: 5


In [ ]:
# Group data by 'Continent' and calculate the mean HDI Rank for each continent
continent_mean_hdi = df.groupby('Continent')['HDI Rank (2021)'].mean()
print("\nMean HDI Rank by Continent:")
print(continent_mean_hdi)


Mean HDI Rank by Continent:
Continent
Africa     153.132075
America     87.628571
Asia        89.187500
Europe      34.642857
Oceania    106.230769
Name: HDI Rank (2021), dtype: float64


In [ ]:
# Perform a t-test to compare HDI Rank between 'High' and 'Low' Human Development Groups
from scipy.stats import ttest_ind

high_hdi_group = df[df['Human Development Groups'] == 'High']['HDI Rank (2021)']
low_hdi_group = df[df['Human Development Groups'] == 'Low']['HDI Rank (2021)']

t_statistic, p_value = ttest_ind(high_hdi_group, low_hdi_group, equal_var=False)
print("\nT-test Results (High vs. Low HDI Groups):")
print("T-statistic:", t_statistic)
print("P-value:", p_value)


T-test Results (High vs. Low HDI Groups):
T-statistic: -32.163498181571434
P-value: 3.903418638838638e-47


 the t-test results indicate that there is a significant difference in the Human Development Index (HDI) Rank between two groups of countries:

The "High" HDI group contains countries with higher levels of human development.
The "Low" HDI group contains countries with lower levels of human development.
The negative t-statistic (-32.16) suggests that, on average, the countries in the "High" HDI group have significantly better (lower) HDI Ranks compared to the countries in the "Low" HDI group.

The extremely small p-value (approximately 3.90e-47) means that this difference is not likely due to random chance; it is statistically significant. the data strongly indicates that there is a meaningful distinction in HDI Rank between these two groups of countries, with the "High" HDI group having a significantly better human development status.